src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs

```diff
@@ -153,6 +153,7 @@ private static bool GetLinqExpressionsBuiltWithIsInterpretingOnly()
         public static bool IsNotIntMaxValueArrayIndexSupported => s_largeArrayIsNotSupported.Value;
 
         public static bool IsAssemblyLoadingSupported => !IsNativeAot;
+        public static bool IsNonBundledAssemblyLoadingSupported => !IsAssemblyLoadingSupported && !IsMonoAOT;
```

Comment:
I don't understand this condition.

Non-bundled assembly loading is supported when assembly loading is not supported and it's not Mono AOT?

This made the test using this run _exclusively_ on NativeAOT. It's not supposed to run there. It's breaking the rolling CI.

I don't understand the specifics, but this probably should have extended the definition of `IsAssemblyLoadingSupported` instead. Or it shouldn't use the `!`. It doesn't make sense as it is.

Reply:
My bad.  The ! needs to go away. I'll put up a fix.

Reply:
https://github.com/dotnet/runtime/pull/73235